packages/client/src/client.ts

```diff
@@ -0,0 +1,303 @@
+      origin: providerOrigin,
+      provider: assertProviderRecord(providerOrigin),
+      requestManifest: getPenumbraManifest(providerOrigin),
+    };
+    void config.requestManifest.then(manifest => {
+      config.provider.addEventListener('penumbrastate', this.providerEventListener);
+      config.parsedManifest = manifest;
+    });
+    return config;
+  }
+
+  /** Assert client is configured. */
+  private assertConfig(providerOrigin?: string) {
+    if (!this.config) {
+      throw new Error('Client not configured');
+    }
+    if (providerOrigin && this.config.origin !== providerOrigin) {
+      throw new Error('Client configuration mismatched');
+    }
+    return this.config;
+  }
+
+  /** Connect to the configured provider. */
+  private createConnection() {
+    const { requestManifest, provider } = this.assertConfig();
+    const request = requestManifest.then(() => provider.connect());
+    return {
+      transport: createChannelTransport({
+        jsonOptions,
+        ...this.transportOptions,
+        getPort: () => request,
+      }),
+      port: request,
+    };
+  }
+
+  /** The promised `MessagePort` returned from this client's provider. */
+  get port(): Promise<MessagePort> | undefined {
+    return this.connection?.port;
+  }
+
+  /** The ConnectRPC `Transport` created by this client's connection. */
+  get transport(): Transport | undefined {
+    return this.connection?.transport;
+  }
+
+  /** The provider origin URI, or `undefined` if this client is not configured
+   * with a provider. */
+  get origin(): string | undefined {
+    return this.config?.origin;
+  }
+
+  /** The boolean provider connection status, or `undefined` if this client is
+   * not configured with a provider. */
+  get connected(): boolean | undefined {
+    return this.config?.provider.isConnected();
+  }
+
+  /** The `PenumbraState` enumerated provider connection state, or `undefined` if
+   * this client is not configured with a provider. */
+  get state(): PenumbraState | undefined {
+    return this.config?.provider.state();
+  }
+
+  /** The parsed `PenumbraManifest` associated with this provider, fetched at
+   * time of client configuration. This will be `undefined` if this client is
+   * not configured with a provider, or if the manifest fetch has not yet
+   * resolved.
+   *
+   * If you hold an active connection to this provider, this should be present.
+   */
+  get manifest(): PenumbraManifest | undefined {
+    return this.config?.parsedManifest;
+  }
+
+  /** Return a promised `PenumbraManifest` associated with this provider, fetched
+   * at time of client configuration, or `undefined` if this client is not
+   * configured with a provider.
+   *
+   * May reject if the manifest is not available. If you hold an active
+   * connection to this provider, this promise should successfully resolve.
+   */
+  fetchManifest(): Promise<PenumbraManifest> | undefined {
```

Comment:
question: is is `public`, `static`, or `private`?

Reply:
unannotated is public, will annotate